internal/bminventory/inventory.go

```diff
@@ -840,13 +841,24 @@ func (b *bareMetalInventory) RegisterAddHostsClusterInternal(ctx context.Context
 func (b *bareMetalInventory) createAndUploadNodeIgnition(ctx context.Context, cluster *common.Cluster, host *models.Host) error {
 	log := logutil.FromContext(ctx, b.log)
 	log.Infof("Starting createAndUploadNodeIgnition for cluster %s, host %s", cluster.ID, host.ID)
-	address := cluster.APIVip
-	if address == "" {
-		address = swag.StringValue(cluster.APIVipDNSName)
-	}
-	ignitionBytes, err := b.IgnitionBuilder.FormatSecondDayWorkerIgnitionFile(address, host.MachineConfigPoolName)
-	if err != nil {
-		return errors.Errorf("Failed to create ignition string for cluster %s", cluster.ID)
+
+	// Specify ignition endpoint based on cluster configuration:
+	var ignitionBytes []byte
+	var err error
+	if cluster.IgnitionEndpointURL != nil {
+		ignitionBytes, err = b.IgnitionBuilder.FormatSecondDayWorkerIgnitionFileByUrl(*cluster.IgnitionEndpointURL, host.IgnitionEndpointToken)
```

Comment:
I think you can add a token pointer parameter to `FormatSecondDayWorkerIgnitionFile` and use that function for both cases.
You will need to create the full ignition URL before calling it though